src/render/tutorials/caching-http-requests/2_0.html.md

```diff
@@ -0,0 +1,314 @@
+Knot.x tasks are some kind of directed graphs consisting of actions (if you want to get more familiar with
+configurable integrations please read [this](https://knotx.io/blog/configurable-integrations/) tutorial). Now, let's get it started 
+and define our task
+
+```hocon
+tasks {
+  api-task {
+    action = fetch-product-with-cache
+    onTransitions {
+      _success {
+        action = product-to-body
+      }
+    }
+  }
+}
+```
+As you see above, to achieve our goal we have to create only one task called `api-task` - this is very simple directed graph.
+The action it performs is `fetch-product-with-cache` which we'll create soon. Then we have `onTransitions`, in our case let's 
+define only `_success` - the action to be performed is `product-to-body`  which we'll also define.
+
+```hocon
+actions {
+  fetch-product-with-cache {
+    factory = in-memory-cache
+    config {
+      cache {
+        maximumSize = 1000
+        ttl = 5000
+      }
+      cacheKey = "product-{param.id}"
+      payloadKey = fetch-product
+    }
+    doAction = fetch-product
+  }
+
+  product-to-body {
+    factory = payload-to-body
+    config {
+      key = fetch-product
+    }
+  }
+
+  fetch-product {
+    factory = http
+    config {
+      endpointOptions {
+        path = /product/id
+        domain = webapi
+        port = 8080
+        allowedRequestHeaders = ["Content-Type"]
+      }
+    }
+  }
+}
+```
+Above you can see all necessary actions definitions. As mentioned before, we've created `fetch-product-with-cache`. The purpose 
+of this action is to add cache functionality. Now you can see how easily we can define custom behaviours for our actions. 
+We've just assigned to it `in-memory-cache` action factory, moreover this action also wraps other - in this case it's 
+`fetch-product` - it's using simple http factory which will call desired endpoint - in our case `/product/id` (we'll define it later).
+Finally we can focus on `product-to-body` - behaviour of this action is really simple - it's just adding previously fetched data
+to response body.
+
+Last thing to do here is to combine this two chunks of code into one file, let's call it `fragments-handler.conf ` and put it
+inside `knotx/conf/routes/handlers/` directory.
+
+## Why we need caching?
+We've just added caching behaviour to action but you may wonder why we did it? Why we're using cache? The general purpose of
+using caching is faster data retrieval - when user call the same endpoint for few times, we don't have to fetch data
+from server for every request - we can cache it and reuse response (if not changed) in the future - it affects in quicker
+response time, what is noticeable by the user. On the other hand, from server side it reduces latency and traffic. Requests
+are satisfied in a shorter time with reused representations.
+
+You may also wonder if we can replace this cache with another one, for example with Redis - yes you can do it very simple!
```

Comment:
provide a link to Redis and rephrase the `You may also wonder` part

Reply:
fixed